### PR TITLE
feat: Add new provider for MS Teams

### PR DESCRIPTION
**Pull request by dcamarasa**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature PR to add MS Teams chat provider.
- **What is the current behavior?** (You can also link to an open issue here)
There is currently no provider implementation for MS Teams yet. Open issue: [1183](https://github.com/novuhq/novu/issues/1183)
- **What is the new behavior (if this is a feature change)?**
Novu users will be able to setup a new integration to MS Teams channels. The only credentials needed is the webhook url itself. MSTeams does not need any other further credentials configuration.
The webhookUrl will be stored on the subscriber entity, the same way it is done in Discord provider.
- **Other information**:
Documentation page on how to retrieve and setup the webhook url in Novu is also added in this PR.
Haven't found any good MS Teams logo with transparent background, so I've added the icon only. Let me know if you want me to add a normal logo with white background as in the gupshup provider.

### CODE REVIEW

File: providers/msTeams/package.json [hunk 1]
```diff
@@ -0,0 +1,88 @@
+{
+  "name": "@novu/msteams",
```
Comment (scopsy) [inline providers/msTeams/package.json:2]:
```suggestion
  "name": "@novu/ms-teams",
```

A small suggestion, could we add a "-" in the package names used here?

**Review by scopsy — COMMENTED** [review]
It looks great @dcamarasa!! This is so exciting 😍

A small thing, could we use `ms-teams` as the folder and package names? So we can be more consistent with the other folder names in the system?

File: providers/msTeams/package.json (hunk 1 above)
Comment (dcamarasa) [reply providers/msTeams/package.json:2]: Sure, I'll change it

Comment (dcamarasa) [reply providers/msTeams/package.json]: Done

**Review by scopsy — APPROVED** [review]
Thank you @dcamarasa this looks perfect!

**Pull request merged**